projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,316 @@
+
+    // Date & Time formats: (Used for files names)
+    // TODO: Check if this should be as part of a "File manager" class.
+    private static final String DATE_TIME_FORMAT = "dd-M-yyyy hh-mm-ss";
+    private static final String DATE_FORMAT = "dd-M-yyyy";
+    private static final String TIME_ZONE = "Asia/Jerusalem";
+    private SimpleDateFormat dateAndTimeFormatter;
+    private SimpleDateFormat dateFormatter;
+
+    // Template's inner text (been read on create from template's file):
+    private String templateContent = "";
+
+// ------------------------------- Lawsuit Form Fields ------------------------------- //
+    // Type: SMS/Phone call/Email.
+    private String spamType = "הודעה אלקטרונית";
+    private String claimAmount = "סכום תביעה";
+    // Date received the spam.
+    private Date receivedSpamDate;
+    // Current date of lawsuit creation (used for lawsuit signature).
+    private Date currentDate;
+    // Indicates whether user has sent 'haser'.
+    private boolean sentHaser = true;
+    // Indicates whether user has sent more then 5 lawsuits already in the past year.
+    private boolean sentMoreThanFiveLawsuits = false;
+    private String moreThanFiveLawsuits ="הגיש";
+    private String lessThanFiveLawsuits = "לא הגיש/ו";
+    private String fiveLawsuitsStatus =""; // Init onCreate().
+    private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    private String claimCaseSubscription = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    // Cases: Didn't subscribe / Sent 'haser' but got spam.
+    private String claimCase = ""; //  Init on fillInTemplate().
+
+    // User:
+    private String userPrivateName = "שם פרטי";
+    private String userLastname = "שם משפחה";
+    private String userID = "ת ז משתמש";
+    private String userAddress = "כתובת משתמש";
+    private String userPhone = "טלפון משתמש";
+    private String userFax = "פקס משתמש";
+
+    // First spam company's details:
+    private String companyName = "שם החברה";
+    private String companyId = "מספר ח.פ";
+    private String companyAddress = "כתובת החברה";
+    private String companyPhone = "טלפון החברה";
+    private String companyFax = "פקס חברה";
+
+    // Second spam company's det ails:
+    private String company2Name = "שם החברה 2";
+    private String company2Id = "מספר ח.פ 2";
+    private String company2Address = "כתובת החברה 2";
+    private String company2Phone = "טלפון החברה 2";
+    private String company2Fax = "פקס חברה 2";
+// ----------------------------- - -------------------------------------------------------:
+    Button createPdfButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_lawsuit_pdf);
+        createPdfButton = (Button) findViewById(R.id.button_createPDF);
+
+        // Check Read/Write permissions:
+        isWriteStoragePermissionGranted();
```

Comment:
Looks like you're not using the boolean return value..

Reply:
Changed to void getWriteStoragePermission().